pkg/vcs/repo.go

```diff
@@ -200,3 +202,47 @@ func GetRepoRef(r *git.Repository, name string) (*plumbing.Reference, error) {
 
 	return nil, fmt.Errorf("failed to get reference: %s", name)
 }
+
+// GetOrgRepos returns full repositories list from the given org.
+func GetOrgRepos(ctx context.Context, client *scm.Client, orgName string) ([]*scm.Repository, error) {
+	opts := scm.ListOptions{Size: 100}
+
+	var list []*scm.Repository
+
+	for {
+		repos, meta, err := client.Organizations.ListRepositories(ctx, orgName, opts)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, src := range repos {
+			if src != nil {
+				list = append(list, src)
+			}
+		}
+
+		opts.Page = meta.Page.Next
+		opts.URL = meta.Page.NextURL
+
+		if opts.Page == 0 && opts.URL == "" {
+			break
+		}
+	}
+
+	return list, nil
+}
+
+// GetOrgFromGitURL parses the organization from the given URL.
+func GetOrgFromGitURL(source string) (string, error) {
```

Comment:
Does it work if it's a multi-part path, like gitlab subgroup?

Reply:
yes， the sub group is tested `https://gitlab.com/terraform-seal-modules/sub`